relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,95 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        scrub_graphql(event);
+
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+}
+
+/// Scrubs GraphQL variables from the event.
+fn scrub_graphql(event: &mut Event) {
+    let mut keys: Option<BTreeSet<String>> = None;
+
+    let mut is_graphql = false;
+
+    // collect the variables keys and scrub them out.
+    if let Some(request) = event.request.value_mut() {
+        if let Some(Value::Object(data)) = request.data.value_mut() {
+            if let Some(api_target) = request.api_target.value() {
+                if api_target.to_lowercase().eq("graphql") {
+                    is_graphql = true;
+                }
+            }
+
+            if is_graphql {
+                if let Some(Annotated(Some(Value::Object(variables)), _)) =
+                    data.get_mut("variables")
+                {
+                    let mut current_keys: BTreeSet<String> = BTreeSet::new();
```

Comment:
If you make `keys` non-optional, you can insert straight into `keys` here, right?